pkg/apps/alert/latest/alertservice.go

```diff
@@ -22,69 +22,46 @@ under the License.
-	})
-
-	service.AddSelectors(map[string]string{"app": util.AlertName, "component": "alert"})
-	service.AddLabels(map[string]string{"app": util.AlertName, "component": "alert"})
-	return service
+// GetAlertClusterService returns a new cluster Service for an Alert
+func (a *SpecConfig) GetAlertClusterService() *components.Service {
+	return util.CreateService(
+		util.GetResourceName(a.alert.Name, util.AlertName, "alert"),
+		map[string]string{"app": util.AlertName, "component": "alert"},
+		a.alert.Namespace,
+		int32(*a.alert.Spec.Port),
+		int32(*a.alert.Spec.Port),
+		horizonapi.ServiceTypeServiceIP,
+		map[string]string{"app": util.AlertName, "component": "alert"},
+	)
 }
 
-// getAlertServiceNodePort returns a new Node Port Service for an Alert
-func (a *SpecConfig) getAlertServiceNodePort() *components.Service {
-	service := components.NewService(horizonapi.ServiceConfig{
-		Name:      util.GetResourceName(a.alert.Name, util.AlertName, "exposed"),
-		Namespace: a.alert.Spec.Namespace,
-		Type:      horizonapi.ServiceTypeNodePort,
-	})
-
-	service.AddPort(horizonapi.ServicePortConfig{
-		Port:       int32(*a.alert.Spec.Port),
-		TargetPort: fmt.Sprintf("%d", *a.alert.Spec.Port),
-		Protocol:   horizonapi.ProtocolTCP,
-		Name:       fmt.Sprintf("%d-tcp", *a.alert.Spec.Port),
-	})
-
-	service.AddSelectors(map[string]string{"app": util.AlertName, "name": a.alert.Name, "component": "alert"})
-	service.AddLabels(map[string]string{"app": util.AlertName, "name": a.alert.Name, "component": "alert"})
-	return service
+// GetAlertServiceNodePort returns a new Node Port Service for an Alert
+func (a *SpecConfig) GetAlertServiceNodePort() *components.Service {
+	return util.CreateService(
+		util.GetResourceName(a.alert.Name, util.AlertName, "exposed"),
+		a.GetLabel("alert"),
+		a.alert.Namespace,
+		int32(*a.alert.Spec.Port),
+		int32(*a.alert.Spec.Port),
+		horizonapi.ServiceTypeNodePort,
+		a.GetLabel("alert"),
+	)
 }
 
-// getAlertServiceLoadBalancer returns a new Load Balancer Service for an Alert
-func (a *SpecConfig) getAlertServiceLoadBalancer() *components.Service {
-	service := components.NewService(horizonapi.ServiceConfig{
-		Name:      util.GetResourceName(a.alert.Name, util.AlertName, "exposed"),
-		Namespace: a.alert.Spec.Namespace,
-		Type:      horizonapi.ServiceTypeLoadBalancer,
-	})
-
-	service.AddPort(horizonapi.ServicePortConfig{
-		Port:       int32(*a.alert.Spec.Port),
-		TargetPort: fmt.Sprintf("%d", *a.alert.Spec.Port),
-		Protocol:   horizonapi.ProtocolTCP,
-		Name:       fmt.Sprintf("%d-tcp", *a.alert.Spec.Port),
-	})
-
-	service.AddSelectors(map[string]string{"app": util.AlertName, "name": a.alert.Name, "component": "alert"})
-	service.AddLabels(map[string]string{"app": util.AlertName, "name": a.alert.Name, "component": "alert"})
-	return service
+// GetAlertServiceLoadBalancer returns a new Load Balancer Service for an Alert
+func (a *SpecConfig) GetAlertServiceLoadBalancer() *components.Service {
```

Comment:
Why the method is changed to public?

Reply:
You're right, I'll make it private, but it seems there is other places where we don't really have any conventions with private vs. public methods.

Reply:
we need to fix it in all places. probably during refractor, we should consider this.

Reply:
OK, a task for 2019.9.0!